components/provisioner/internal/provisioning/service.go

```diff
@@ -475,6 +459,28 @@ func (r *service) RollBackLastUpgrade(runtimeID string) (*gqlschema.RuntimeStatu
 	return r.RuntimeStatus(runtimeID)
 }
 
+func (r *service) isKyma2Installed(cluster model.Cluster) bool {
+
+	if util.IsNilOrEmpty(cluster.ActiveKymaConfigId) {
+		return true
+	}
+
+	if cluster.Kubeconfig == nil {
+		log.Warnf("Kubeconfig for cluster %s is missing", cluster.ID)
+		return false
```

Comment:
When there is no kubeconfig, should we proceed with the "uninstall" flow? There is nothing to uninstall :D

Reply:
This will be processed correctly in trigger_kyma_uninstall.go step. Uninstall operation will be skipped if there is no kubeconfig. My intention was to restore original implementation.